Thekra_Alowais/week_04/day_02/mtaRuby.rb

```diff
@@ -0,0 +1,55 @@
+$lines = {
+  n: ["Times Square", "34th", "28th", "23rd", "Union Square", "8th"],
+  l: ["8th", "6th", "Union Square", "3rd", "1st"],
+  s: ["Grand Central", "33rd", "28th", "23rd", "Union Square", "Astor Place"],
+}
+$stopsCount = 0
+
+def getLine(key)
+  $lines[key]
+end
+
+# p getLine(:n)
+def getTripString(line, startStop, endStop)
+  #   p "getTripString methods #{line} +  #{startStop}  + #{endStop}"
+  line_array = getLine(line)
+  #   p "#{line_array}"
+  stopsString = ""
+  startIndex = line_array.index(startStop)
+  endIndex = line_array.index(endStop)
+  p "startIndex #{startIndex} and endIndex #{endIndex}"
+  if startIndex < endIndex
+    i = startIndex
+    while startIndex <= endIndex
```

Comment:
So the problem here is that you're updating `i`, but your `while condition` is using `startIndex` to check.